ucum-java8/src/main/java/systems/uom/ucum/format/UCUMFormat.java

```diff
@@ -207,24 +207,48 @@ public Appendable format(Unit<?> unknownUnit, Appendable appendable)
 		    temp.append(')');
 		}
 		int pow = productUnits.get(u);
-		if (app.length() > 0) { // Not the first unit.
-		    if (pow >= 0) {
-			app.append('.');
-		    } else {
-			app.append('/');
-			pow = -pow;
-		    }
+				int indexToAppend;
+				if(app.length() > 0) { // Not the first unit.
+
+					if(pow >= 0) {
+
+						if(app.indexOf("1/") >= 0) {
+							indexToAppend = app.indexOf("1/");
```

Comment:
Regarding my comment about `final`, I see that the variable was declared above. Still if you _do_ declare it final, the compile could make sure you initialize it --- but I don't know if you later change it, so that's just a suggestion to look into.